lib/Dialect/AIE/IR/AIETargetModel.cpp

```diff
@@ -647,6 +652,21 @@ bool AIE2TargetModel::isLegalTileConnection(int col, int row,
   return false;
 }
 
+bool AIE2TargetModel::hasProperty(ModelProperty prop) const {
+  uint64_t properties = 0;
+  properties |= AIETargetModel::UsesSemaphoreLocks;
+  properties |= AIETargetModel::UsesMultiDimensionalBDs;
+  properties |= AIETargetModel::IsNPU;
+  return (properties & prop) == prop;
+}
```

Comment:
AIE2 does not imply `IsNPU`. It should get inherited from `BaseNPUTargetModel` or `BaseNPU3TargetModel`.

Reply:
Thanks for finding the bug. Fixed.